NASA-PDS/portal-wp
# Add Search Page

## 🗒️ Summary
This PR adds the search page to portal-wp. 

What is left to do but is not yet ready. Will be additional tickets in the near future.
- Set the urls for result links. Need the links.
- Missing designs for result types Facility, Document, Resource, instrument_host. Designs are being drawn up.
- Page type filters. Need indexer update.

## ⚙️ Test Data and/or Report
Get the latest wds develop branch https://github.com/NASA-PDS/wds/tree/develop
Get the latest wds-react develop branch at https://github.com/NASA-PDS/wds-react/tree/develop

Follow portal-wp frontend dev instructions https://github.com/NASA-PDS/portal-wp/tree/features/prototype-search-page?tab=readme-ov-file#frontend-service-workaround

Go to http://localhost:5173/search/
Enter something in the search box. See the results appear. 

Other testable functions:

- Un/click a filter
- Filter the filters by entering text in the filter text box.
- See the result numbers change in the top left.
- Expand/collapse all will show or hide result details.
- Number of results in the options will change the number of results.
- Relevance will change the result order.
- Pagination buttons at the bottom will change page.
- Copy pasting the url will open with the same search and filters applied.
- Change window size to see responsiveness.
- Search for unfound strings to see the no results message. EX: lakjsdklfajdsklfjaklsdjfklajdsfk
- run `npm run build`. Verify that the project is built with no errors.

## ♻️ Related Issues
Pull request for wds https://github.com/NASA-PDS/wds/pull/52
Pull request for wds-react https://github.com/NASA-PDS/wds-react/pull/60
Epic for search page https://github.com/NASA-PDS/portal-wp-tasks/issues/58
